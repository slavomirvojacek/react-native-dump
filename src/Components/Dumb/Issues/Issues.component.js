import React from 'react'
import { View } from 'react-native'

import Config from './Issues.component.config'

import Issue from '../Issue/Issue.component'

const Issues = ({issues = [], ...rest}) => {
  const Children = issues
    .map(issue => {
      const issueProps = {issue, ...rest}
      return <Issue key={issue.id} {...issueProps}/>
    })

  return <View>{Children}</View>
}

Issues.displayName = Config.displayName
Issues.propTypes = Config.propTypes

export default Issues
