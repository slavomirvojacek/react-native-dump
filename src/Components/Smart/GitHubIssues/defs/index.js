const userDef = (id = '') => ({id})

const reactionDef = (override = {}) => {
  const def = {
    id: '',
    content: '',
    user: userDef()
  }

  return {...def, ...override}
}

const issueDef = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: []
  }

  return {...def, ...override}
}

export {
  userDef, reactionDef, issueDef
}
