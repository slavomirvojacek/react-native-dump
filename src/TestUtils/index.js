export { assertGql } from './gql'
export { mockSetState } from './mocking'
export { assertSnapshots } from './snapshot'
