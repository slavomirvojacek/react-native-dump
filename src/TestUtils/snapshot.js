import React from 'react'
import renderer from 'react-test-renderer'

import { mockSetState } from './mocking'

const assertSnapshot = (C, config) => {
  const {props, state, desc, throws} = config

  if (throws) {
    test(desc, () => {
      expect(() => {
        if (C.prototype.render) {
          new C(props)
        } else {
          C(props)
        }
      }).toThrow(throws)
    })
  } else {
    let Component
    let RenderedComponent

    if (C.prototype.render) {
      const CWithMockState = mockSetState(C)
      Component = new CWithMockState(props)

      if (state) {
        Component.setState(state)
      }

      RenderedComponent = Component.render()
    } else {
      Component = C(props)
      RenderedComponent = Component
    }

    test(desc, () => {
      const tree = renderer
        .create(RenderedComponent)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
}

const assertSnapshots = (Component, configs) => {
  configs.forEach(config => assertSnapshot(Component, config))
  // if (Component.gql) assertGql(Component)
}

export { assertSnapshots }
