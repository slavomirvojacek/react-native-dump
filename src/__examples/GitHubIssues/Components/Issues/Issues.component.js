import React from 'react'
import { View, Text } from 'react-native'

import { Reactions } from '../Reactions/Reactions.component'
// @TODO move to its own file here
import styles from './Issues.component.styles'

import { noop } from '../../../../Utils/misc'

const Issues = ({issues = [], openIssue = noop, ...rest}) => {
  const Children = issues
    .map(issue => {
      const onPress = () => openIssue(issue.url)
      const reactions = issue.reactions.nodes
      const reactionsProps = {reactions, issue, ...rest}

      return (
        <View key={issue.id}>
          <Text onPress={onPress} style={styles.issueLink}>{issue.title}</Text>
          <Reactions {...reactionsProps} />
        </View>
      )
    })

  // @TODO figure out whether View is necessary
  return <View style={{backgroundColor: 'red'}}>{Children}</View>
}

export default Issues
