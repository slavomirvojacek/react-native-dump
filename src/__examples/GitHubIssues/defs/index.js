const userDef = (id = '') => ({id})

const reactionDef = ({
                       id = '',
                       content = '',
                       user = userDef()
                     }) => ({
  id, content, user
})

const issueDef = ({
                    id = '',
                    title = '',
                    url = '',
                    bodyText = '',
                    reactions = []
                  }) => ({
  id, title, url, bodyText, reactions
})

export {
  userDef, reactionDef, issueDef
}
