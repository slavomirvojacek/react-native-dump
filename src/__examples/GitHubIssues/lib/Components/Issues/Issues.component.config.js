import PropTypes from 'prop-types'

const displayName = 'IssuesComponent'

const propTypes = {
  issues: PropTypes.array.isRequired,
}

// Only used for testing
const testDefaultProps = {
  issues: [],
}

export default {
  displayName, propTypes, testDefaultProps
}
