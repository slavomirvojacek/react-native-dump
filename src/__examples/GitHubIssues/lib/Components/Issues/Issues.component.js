import React from 'react'
import { View, Text } from 'react-native'

import Config from './Issues.component.config'
import styles from './Issues.component.styles'

import { Reactions } from '../Reactions/Reactions.component'
import { openUrl } from '../../../../../Utils/misc'

const Issues = ({issues = [], ...rest}) => {
  const Children = issues
    .map(issue => {
      const onPress = () => openUrl(issue.url)
      const reactions = issue.reactions.nodes
      const reactionsProps = {reactions, issue, ...rest}

      return (
        <View key={issue.id}>
          <Text onPress={onPress} style={styles.issueLink}>{issue.title}</Text>
          <Reactions {...reactionsProps} />
        </View>
      )
    })

  // @TODO figure out whether View is necessary
  return <View style={{backgroundColor: 'red'}}>{Children}</View>
}

Issues.displayName = Config.displayName
Issues.propTypes = Config.propTypes
// Issues.defaultProps = Config.defaultProps

export default Issues
