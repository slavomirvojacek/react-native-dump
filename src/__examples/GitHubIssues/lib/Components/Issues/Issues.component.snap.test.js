import Issues from './Issues.component'
import Config from './Issues.component.config'

import { issueDef } from '../../../defs'
import { assertSnapshots } from '../../../../../TestUtils/snapshot'

jest
  .mock('../Issue/Issue.component', () => 'Issue')

describe('Issues Component Snapshots', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  const configs = [
    {
      props: Config.testDefaultProps,
      desc: 'renders correctly'
    },
    {
      props: {
        ...Config.testDefaultProps,
        issues: [issueDef()],
      },
      desc: 'renders correctly when issues present'
    },
  ]

  assertSnapshots(Issues, configs)
})
