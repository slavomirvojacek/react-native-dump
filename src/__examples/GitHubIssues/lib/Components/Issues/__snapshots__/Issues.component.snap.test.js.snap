// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issues Component Snapshots renders correctly 1`] = `<View />`;

exports[`Issues Component Snapshots renders correctly when issues present 1`] = `
<View>
  <Issue
    issue={
      Object {
        "bodyText": "",
        "id": "",
        "reactions": Array [],
        "title": "",
        "url": "",
      }
    }
  />
</View>
`;
